pkg/observable/interface.go

```diff
@@ -11,6 +11,7 @@ import "context"
 // notified of new values asynchronously.
 // It is analogous to a publisher in a "Fan-Out" system design.
 type Observable[V any] interface {
+	Next(context.Context) V
```

Comment:
Either add a comment or what this does or add a TODO so we tend to it in the future.

Reply:
Thanks for calling this out! :raised_hands: 

I added godoc comments to all these.